test/app-tap/gh-2717-no-quit-sigint.test.lua

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env tarantool
+
+local tap = require('tap')
+local popen = require('popen')
+local process_timeout = require('process_timeout')
+local fio = require('fio')
+local clock = require('clock')
+local fiber = require('fiber')
+
+--
+-- gh-2717: tarantool console quit on sigint.
+--
+local file_read_timeout = 60.0
+local file_read_interval = 0.2
+local file_open_timeout = 60.0
+local prompt = 'tarantool> '
+
+local TARANTOOL_PATH = arg[-1]
+local test = tap.test('gh-2717-no-quit-sigint')
+
+test:plan(6)
+local cmd = 'ERRINJ_STDIN_ISATTY=1 ' .. TARANTOOL_PATH .. ' -i 2>&1'
+local ph = popen.new({cmd}, {
+    shell = true,
+    setsid = true,
+    group_signal = true,
+    stdout = popen.opts.PIPE,
+    stderr = popen.opts.DEVNULL,
+    stdin = popen.opts.PIPE,
+})
+assert(ph, 'process is not up')
+
+local start_time = clock.monotonic()
+local deadline = 10.0
```

Comment:
Nit: I would call it `time_quota` or kinda, becase `deadline` in my understanding is a point of time, not a time period.

Reply:
Fixed.